TPP.Core.Tests/Commands/Definitions/BettingCommandsTest.cs

```diff
@@ -0,0 +1,118 @@
+using TPP.Core.Commands;
+using TPP.Core.Commands.Definitions;
+using TPP.Match;
+using TPP.Persistence.Models;
+
+namespace TPP.Core.Tests.Commands.Definitions
+{
+    public class BettingCommandsTest
+    {
+        private static User MockUser(string name) => new User(
+            id: Guid.NewGuid().ToString(),
+            name: name, twitchDisplayName: "☺" + name, simpleName: name.ToLower(), color: null,
+            firstActiveAt: Instant.FromUnixTimeSeconds(0), lastActiveAt: Instant.FromUnixTimeSeconds(0),
+            lastMessageAt: null, pokeyen: 0, tokens: 0,
+            selectedBadge: null);
+
+        private static Message MockMessage(User user, MessageSource source = MessageSource.Chat) =>
+            new(user, string.Empty, source, string.Empty);
+
+        private static readonly ArgsParser ArgsParser;
+
+        static BettingCommandsTest()
+        {
+            ArgsParser = new ArgsParser();
+            ArgsParser.AddArgumentParser(new PositiveIntParser());
+            ArgsParser.AddArgumentParser(new PokeyenParser());
+            ArgsParser.AddArgumentParser(new PercentageParser());
+            ArgsParser.AddArgumentParser(new SideParser());
+            ArgsParser.AddArgumentParser(new OneOfParser(ArgsParser));
+        }
+
+        [TestFixture]
+        public class Bet
+        {
+            [Test]
+            public async Task deny_if_betting_not_available()
+            {
+                var commands = new BettingCommands(() => null);
+
+                CommandResult result = await commands.Bet(new CommandContext(MockMessage(MockUser("me")),
+                    ImmutableList.Create("!bet 1 blue"), ArgsParser));
+
+                Assert.AreEqual("betting not available right now", result.Response);
+            }
+
+            [Test]
+            public async Task deny_if_betting_closed()
+            {
+                Mock<IBettingPeriod<User>> bettingPeriodMock = new();
+                bettingPeriodMock.Setup(b => b.IsBettingOpen).Returns(false);
+                var commands = new BettingCommands(() => bettingPeriodMock.Object);
+
+                CommandResult result = await commands.Bet(new CommandContext(MockMessage(MockUser("me")),
+                    ImmutableList.Create("!bet 1 blue"), ArgsParser));
+
+                Assert.AreEqual("betting is already closed", result.Response);
+            }
+
+            [Test]
+            public async Task only_allow_from_chat()
+            {
+                User user = MockUser("me");
+                Mock<IBettingPeriod<User>> bettingPeriodMock = new();
+                var commands = new BettingCommands(() => bettingPeriodMock.Object);
+
+                CommandResult result = await commands.Bet(new CommandContext(
+                    MockMessage(user, source: MessageSource.Whisper),
+                    ImmutableList.Create("100", "blue"), ArgsParser));
+
+                Assert.AreEqual("You may only bet through chat", result.Response);
```

Comment:
worth noting that some modes allow betting through whispers, might be worth adding the framework for that now

Reply:
hmm, that's mentally in the big blob of gimmicks/modes/metas for which I don't know how to implement them yet